server/auth/types/multiple/multi_auth.ts

```diff
@@ -0,0 +1,179 @@
+/*
+ *   Copyright OpenSearch Contributors
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+import {
+  CoreSetup,
+  SessionStorageFactory,
+  IRouter,
+  ILegacyClusterClient,
+  OpenSearchDashboardsRequest,
+  Logger,
+  LifecycleResponseFactory,
+  AuthToolkit,
+} from '../../../../opensearch-dashboards/server';
+import { OpenSearchDashboardsResponse } from '../../../../../../src/core/server/http/router';
+import { SecurityPluginConfigType } from '../../..';
+import { AuthenticationType, IAuthenticationType } from '../authentication_type';
+import { ANONYMOUS_AUTH_LOGIN, AuthType, LOGIN_PAGE_URI } from '../../../../common';
+import { composeNextUrlQueryParam } from '../../../utils/next_url';
+import { MultiAuthRoutes } from './routes';
+import { SecuritySessionCookie } from '../../../session/security_cookie';
+import { BasicAuthentication, OpenIdAuthentication, SamlAuthentication } from '../../types';
+
+export class MultipleAuthentication extends AuthenticationType {
+  private authTypes: string | string[];
+  private authHandlers: Map<string, AuthenticationType>;
+
+  constructor(
+    config: SecurityPluginConfigType,
+    sessionStorageFactory: SessionStorageFactory<SecuritySessionCookie>,
+    router: IRouter,
+    esClient: ILegacyClusterClient,
+    coreSetup: CoreSetup,
+    logger: Logger
+  ) {
+    super(config, sessionStorageFactory, router, esClient, coreSetup, logger);
+    this.authTypes = this.config.auth.type;
+    this.authHandlers = new Map<string, AuthenticationType>();
+  }
+
+  public async init() {
+    const routes = new MultiAuthRoutes(this.router, this.sessionStorageFactory);
+    routes.setupRoutes();
+
+    for (let i = 0; i < this.authTypes.length; i++) {
+      switch (this.authTypes[i].toLowerCase()) {
+        case AuthType.BASIC: {
+          const BasicAuth = new BasicAuthentication(
+            this.config,
+            this.sessionStorageFactory,
+            this.router,
+            this.esClient,
+            this.coreSetup,
+            this.logger
+          );
+          await BasicAuth.init();
```

Comment:
why not using `createAuthentication`?

Reply:
The return value type for `createAuthentication` is `IAuthenticationType`, which only has type, authHandler and init can be called directly. However, mult_auth need to save `AuthenticationType` in the `authHandlers` Map in order to reuse the methods for each corresponding auth type.